cartoframes/context.py

```diff
@@ -1585,9 +1591,14 @@ def data(self, table_name, metadata, persist_as=None, how='the_geom'):
                 tablename=table_name,
                 rowid='cartodb_id' if how == 'the_geom' else how,
                 cols=cols,
+                table_cols=','.join('t.{}'.format(c) for c in table_columns),
                 meta=_meta.to_json(orient='records').replace('\'', '\'\''))
-        return self.query(query,
-                          table_name=persist_as)
+
+        result = self.fetch(query)
+        if persist_as:
+            self.write(result, persist_as, overwrite=True)
```

Comment:
Maybe we should be smarter and if you need to store it, just do a Batch API `SELECT INTO` and then download the result (or not even download the result)?

Reply:
Makes sense, I've added a `Dataset.from_query` method that creates a table from the query. Then we use `Dataset.download` to get the result.